src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -349,9 +356,17 @@
   [globals]
   (select-keys globals [:scf-read-db :warn-experimental :url-prefix :pretty-print]))
 
+(defn valid-query?
+  [version query-map]
+  (let [{:keys [remaining-query entity query-options]} (qeng/user-query->engine-query version query-map)
+        validated-query (qeng/query->sql remaining-query entity version query-options)]
```

Comment:
I'd push this into the body of the let. When I see something let bound, I am expecting to see it used either in the let or in the body, which `validated-query` is not.